test/test_fx_experimental.py

```diff
@@ -0,0 +1,68 @@
+import torch
+from collections import namedtuple
+from torch.fx.symbolic_trace import symbolic_trace
+from torch.fx.experimental import GraphManipulation
+from torch.fx.experimental.Partitioner import Partitioner, DAG
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.jit_utils import JitTestCase
+
+
+class TestFX(JitTestCase):
+    def test_find_single_partition(self):
+        class TestModule(torch.nn.Module):
+            def forward(self, a, b):
+                return a + b
+        m = TestModule()
+        traced = symbolic_trace(m)
+        GraphManipulation.get_size_of_all_nodes(
+            traced,
+            [torch.rand(1), torch.rand(2)]
+        )
+        partitioner = Partitioner()
+        device = namedtuple('device', ['name', 'available_mem_bytes'])
+        devices = [
+            device('dev_0', 125),
+            device('dev_1', 125),
+            device('dev_2', 125)
+        ]
+        ret = partitioner.partition_graph(traced, m, devices)
+        module_with_submodules = ret.module_with_submodules
+        a = torch.rand(1)
+        b = torch.rand(1)
+        self.assertEqual(traced(a, b), module_with_submodules(a, b))
+
+    def test_size_based_partition(self):
+        class TestModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear_mod_0 = torch.nn.Linear(3, 4)
+                self.linear_mod_1 = torch.nn.Linear(4, 4)
+            def forward(self, a, b, c):
+                d = self.linear_mod_0(a)
+                e = self.linear_mod_1(b)
+                f = torch.rand(4)
+                g = torch.rand(5)
+                return c + d + e, f, g
+
+        m = TestModule()
+        traced = symbolic_trace(m)
+        GraphManipulation.get_size_of_all_nodes(
+            traced,
+            [torch.rand(3), torch.rand(4), torch.rand(4)]
+        )
+        partitioner = Partitioner()
+        device = namedtuple('device', ['name', 'available_mem_bytes'])
+        devices = [
+            device('dev_0', 125),
+            device('dev_1', 125),
+            device('dev_2', 125)
+        ]
+        ret = partitioner.partition_graph(traced, m, devices)
```

Comment:
partitioner.partition_graph return a namedtuple with two things: a DAG with all info on nodes of base graph module and a new graph module with submodules